src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -411,17 +412,44 @@ private String findRubyHome() throws IOException {
             return home.getCanonicalPath();
         }
 
+        StringBuilder warning = new StringBuilder("TruffleRuby's home was not explicitly set.\n");
+
         if (!options.LAUNCHER.isEmpty()) {
-            final File candidate = Paths.get(options.LAUNCHER).getParent().getParent().toFile();
+            final Path canonicalLauncherPath = Paths.get((new File(options.LAUNCHER)).getCanonicalPath());
```

Comment:
Double parens here around `new File`.

Reply:
Are you sure? I don't see it.

Reply:
`Paths.get((new File(options.LAUNCHER)).getCanonicalPath());`
`Paths.get(new File(options.LAUNCHER).getCanonicalPath());`

Reply:
ah it's not double parents but one too many.